code/game/objects/items/storage/storage.dm

```diff
@@ -25,7 +25,7 @@
 	var/collection_mode = 1;  //0 = pick one at a time, 1 = pick all on tile, 2 = pick all of a type
 	var/preposition = "in" // You put things 'in' a bag, but trays need 'on'.
 	var/rustle_jimmies = TRUE	//Play the rustle sound on insertion
-
+	var/block_open_while_equipped = 0 // Can we open this storage container while we're wearing it?

```

Comment:
Stylistic thing I'll nitpick about before Purpose shows up:

Use `FALSE/TRUE` instead of `0/1`